rotkehlchen/chain/evm/decoding/aave/common.py

```diff
@@ -206,18 +211,20 @@ def _decode_withdrawal(
                 )
             ):
                 if (
-                    event.location_label == to and
+                    (event.location_label == to or (event.asset == self.evm_inquirer.native_token and is_weth_user)) and  # noqa: E501
                     event.event_type == HistoryEventType.RECEIVE and
                     event.address != ZERO_ADDRESS
                 ):
                     event.event_type = HistoryEventType.WITHDRAWAL
                     event.event_subtype = HistoryEventSubType.REMOVE_ASSET
-                    event.location_label = user
+                    if is_weth_user is False:  # else location label is correctly set by the transfer decoder  # noqa: E501
+                        event.location_label = user
```

Comment:
empty newline for readability

```suggestion
                        event.location_label = user

```